src/base/bittorrent/session.cpp

```diff
@@ -1808,16 +1840,16 @@ void Session::processShareLimits()
             }
 
             if (torrent->seedingTimeLimit() != TorrentHandle::NO_SEEDING_TIME_LIMIT) {
-                const int seedingTimeInMinutes = torrent->seedingTime() / 60;
-                int seedingTimeLimit = torrent->seedingTimeLimit();
+                const std::chrono::minutes seedingTime = std::chrono::duration_cast<std::chrono::minutes>(torrent->seedingTime());
+                std::chrono::minutes seedingTimeLimit = torrent->seedingTimeLimit();
```

Comment:
In fact, I would suggest not using the `minutes` type for any variables or data members, instead use the `duration` type of the clock you're using. for example: `lt::duration` (I realize you can't use that directly, since it wasn't there in 1.1.x). But for instance, you could make a typedef for `std::chrono::monotonic_clock::duration` and use that. All the unit conversions will be done (mostly) automatically

Reply:
Certain durations in qBt have resolution of minutes explicitly. I would not want tot loose that.